lib/puppet/module.rb

```diff
@@ -67,11 +67,23 @@ def initialize(name, path, environment)
 
     load_metadata if has_metadata?
 
-    validate_puppet_version
-
     @absolute_path_to_manifests = Puppet::FileSystem::PathPattern.absolute(manifests)
   end
 
+  # This is here because puppetversion as module data is deprecated.
+  def puppetversion
```

Comment:
Since this is nonfunctional and should probably be removed in the next breaking release, this should be documented as deprecated with something like `# @deprecated The puppetversion module metadata field is no longer used`